System/SpatialPartitionGrid.cpp

```diff
@@ -35,36 +35,32 @@ namespace RTE {
 		m_Height = reference.m_Height;
 		m_CellSize = reference.m_CellSize;
 		m_Cells = reference.m_Cells;
-		m_UsedCells = reference.m_UsedCells;
+		m_UsedCellIds = reference.m_UsedCellIds;
 		return 0;
 	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	void SpatialPartitionGrid::Reset() {
-		const Activity* activity = g_ActivityMan.GetActivity();
-		RTEAssert(activity, "Spatial partition grid with no running activity!");
+		const Activity *activity = g_ActivityMan.GetActivity();
+		RTEAssert(activity, "Tried to reset spatial partition grid with no running activity!");
 
 		for (int team = Activity::NoTeam; team < Activity::MaxTeamCount; ++team) {
-			bool teamActive = team == Activity::NoTeam || activity->TeamActive(team);
-			if (!teamActive) {
-				continue;
-			}
-
-			for (int usedCell : m_UsedCells) {
-				MOIDList& moidList = m_Cells[team+1][usedCell];
-				moidList.clear();
+			if (team == Activity::NoTeam || activity->TeamActive(team)) {
```

Comment:
Ugh I hate nesting. I find this form of code much more difficult to follow, whereas the original is very clear to me. I’d much prefer if this was reverted.

I like using continues to both avoid nesting, but also make it clear that the following code is an exceptional case.

Reply:
I think this may be a you-specific thing haha. Cause this is literally just straightforward logic if you read through it (it's 4 lines of code + brackets):
```
for each team
    if team is no team or team is active
        do the thing
```
vs
```
for each team
    if !team is active
        continue

    do the thing
```
I didn't change it for the Add call, because I think the more complex check makes it cleaner and clearer with the continue there, but I don't really see how the continue is cleaner and clearer in this case.